doc/src/lexextensions.md

```diff
@@ -0,0 +1,59 @@
+# Lex extensions
+    
+Flags can be specified at compile time through `LexFlags` or at `.l` file parse time using
+a `%grmtools{ }` section. At compile time these flags can be enabled using
+[`CTLexerBuilder`](https://docs.rs/lrlex/latest/lrlex/struct.CTLexerBuilder.html) methods.
+
+Flags commonly affect the parsing of the lex file, the interpretation regular expressions,
+and set limits.
+
+Boolean flags are specified by their name, and can be negated by prefixing with `!`
+other flags should specify their value immediately after the flag name.
+
+## Example
+```
+%grmtools {
+    allow_wholeline_comments
+    !octal
+    size_limit 1024
+}
+%%
+. "rule"
+```
+## List of flags:
+
+* `posix_escapes: bool` <p>
```

Comment:
Perhaps this would be better as a table, so it could have the flag name, and arg in separate cells.

Reply:
Should be fixed in [7c21a9d](https://github.com/softdevteam/grmtools/pull/499/commits/7c21a9df013bb73600cdd8c7bd647c943db92cf6)

Reply:
Formatting this sort of thing, especially in Markdown, is always a challenge. I think a table is not a bad idea but the "extra description rows" confused me at first. One way around this might be something like:

```
| `allow_wholeline_comment` [*] | bool  | &cross;  | &cross;       |
...

[*] Enables c++ style `// comments` at the start of lines. Note: The `/` character may require escaping when used in a regex.
```

Maybe using † instead of `[*]` or whatnot?

Reply:
mdbook doesn't actually seem to preserve the footnote character unfortunately, but should be fixed in [62b616f](https://github.com/softdevteam/grmtools/pull/499/commits/62b616f090e397c60f57c54a1d3802c8fc2a0e90)